app/src/main/java/com/example/taskyy/data/repositories/AgendaRepositoryImpl.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.taskyy.data.repositories
+
+import android.util.Log
+import com.example.taskyy.data.remote.TaskyyApi
+import com.example.taskyy.domain.repository.AgendaRepository
+import retrofit2.HttpException
+import javax.inject.Inject
+
+class AgendaRepositoryImpl @Inject constructor(
+    private val retrofit: TaskyyApi,
+    ): AgendaRepository {
+
+    override suspend fun logout(): Boolean {
+        var wasLogoutSuccessful = true
+        try {
+            retrofit.logoutUser()
+        } catch (e: HttpException) {
+            Log.e("Tag", e.code().toString())
+            wasLogoutSuccessful = false
+        }
+        return wasLogoutSuccessful
```

Comment:
You can also use try/catch as an expression and return it :)
```
return try {
    retrofit.logoutUser()
    true
} catch(e: HttpException) {
    false
}
```
Also, with Retrofit it's important to catch `IOException` in addition to `HttpException`. They're thrown when there's no internet connection for example or when the JSON parsing fails